backend/index.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express')
+const mongoose = require('mongoose')
+
+const config = require('./config')
+const https = require('./https')
+const Modul = require('./Moses/Modul')
+const app = express()
+const conn = mongoose.connect(config.network.mongoURI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    useFindAndModify: false,
+    useCreateIndex: true
+})
+
+const MosesModule = mongoose.model('MosesModule', new mongoose.Schema({
+    nummer: { type: Number },
+    titel: { type: String },
+    lernergebnisse: { type: String },
+    lehrinhalte: { type: String },
+    fakultaet: { type: String },
+    sekretariat: { type: String },
+    institut: { type: String },
+    fachgebiet: { type: String },
+    verantwortlich: { type: String },
+    ansprechpartner: { type: String },
+    email: { type: String }
+}));
+
+let modules;
+https(config.resources.moses.getLinkToAllModules(), (data) => { modules = Modul.getModules(data) })
+
+app.get('/moses', function (req, res) {
+    res.send(modules)
+})
+
+app.get('/moses/:id', function (req, res) {
```

Comment:
Funktion zu groß -> nicht lesbar
Kann in mehrere Teilfunktionen aufgeteilt werden, wobei jede Funktion nur eine Sache erledigt (Single Responsibility Principle)

Reply:
zu viele Bedingungen verschachtelt

Reply:
da es callbacks sind ist es nicht trivial möglich die funktionen aufzuteilen. der code würde dadurch noch unsauberer sein.